packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -81,194 +109,271 @@ ModalBody.propTypes = {
-        return React.cloneElement(child, {
-          closeModal: closeModal,
-          inputref: button,
+    function handleBlur({
+      target: oldActiveNode,
+      relatedTarget: currentActiveNode,
+    }) {
+      if (open && currentActiveNode && oldActiveNode && innerModal.current) {
+        const { current: bodyNode } = innerModal;
+        const { current: startSentinelNode } = startSentinel;
+        const { current: endSentinelNode } = endSentinel;
+        wrapFocus({
+          bodyNode,
+          startTrapNode: startSentinelNode,
+          endTrapNode: endSentinelNode,
+          currentActiveNode,
+          oldActiveNode,
+          selectorsFloatingMenus,
         });
-      default:
-        return child;
+      }
     }
-  });
 
-  useEffect(() => {
-    if (prevOpen !== isOpen) {
-      toggleClass(document.body, `${prefix}--body--with-modal-open`, isOpen);
+    function closeModal(evt) {
+      if (!onClose || onClose(evt) !== false) {
+        setisOpen(false);
+      }
     }
-  });
-
-  useEffect(() => {
-    return () =>
-      toggleClass(document.body, `${prefix}--body--with-modal-open`, false);
-  }, []); // eslint-disable-line react-hooks/exhaustive-deps
-
-  useEffect(() => {
-    toggleClass(document.body, `${prefix}--body--with-modal-open`, open);
-  }, [open, prefix]);
-
-  useEffect(() => {
-    const focusButton = (focusContainerElement) => {
-      if (focusContainerElement) {
-        const primaryFocusElement =
-          focusContainerElement.querySelector(selectorPrimaryFocus);
-        if (primaryFocusElement) {
-          primaryFocusElement.focus();
-          return;
+
+    const modalClass = cx(
+      `${prefix}--modal`,
+      isOpen && 'is-visible',
+      customClassName,
+      danger && `${prefix}--modal--danger`
+    );
+
+    const containerClass = cx(
+      `${prefix}--modal-container`,
+      size && `${prefix}--modal-container--${size}`,
+      isFullWidth && `${prefix}--modal-container--full-width`,
+      containerClassName
+    );
+
+    // Generate aria-label based on Modal Header label if one is not provided (L253)
+    let generatedAriaLabel;
+    const childrenWithProps = React.Children.toArray(children).map((child) => {
+      switch (true) {
+        case isElement(child) &&
+          child.type === React.createElement(ModalHeader).type: {
+          const el = child as ReactElement<
+            ModalHeaderProps,
+            typeof ModalHeader
+          >;
+          generatedAriaLabel = el.props.label;
+          return React.cloneElement(el, { closeModal });
         }
-        if (button.current) {
-          button.current.focus();
+
+        case isElement(child) &&
+          child.type === React.createElement(ModalFooter).type: {
+          const el = child as ReactElement<
+            ModalFooterProps,
+            typeof ModalFooter
+          >;
+          return React.cloneElement(el, { closeModal, inputref: button });
```

Comment:
Ditto about the exact props shape, including the (now fixed) omission of event parameter in the `ModalFooter` props